src/overlays/actors/ovl_En_Time_Tag/z_en_time_tag.c

```diff
@@ -42,40 +42,268 @@ const ActorInit En_Time_Tag_InitVars = {
+        ActorCutscene_StartAndSetUnkLinkFields(this->actor.cutscene, &this->actor);
+        this->actionFunc = func_80AC9FD4;
+        gSaveContext.unk_3DD0[3] = 0;
+        if (CHECK_QUEST_ITEM(QUEST_REMAINS_ODOWLA) && CHECK_QUEST_ITEM(QUEST_REMAINS_GOHT) &&
+            CHECK_QUEST_ITEM(QUEST_REMAINS_GYORG) && CHECK_QUEST_ITEM(QUEST_REMAINS_TWINMOLD)) {
+            gSaveContext.save.weekEventReg[25] |= 2;
+        }
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Time_Tag/func_80ACA268.s")
+void func_80ACA0A8(EnTimeTag* this, GlobalContext* globalCtx) {
+    EnTimeTag* this2 = this;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Time_Tag/func_80ACA348.s")
+    if ((globalCtx->msgCtx.ocarinaMode == 3) && (globalCtx->msgCtx.unk1202E == 4)) {
+        if (this->actor.cutscene != -1) {
+            this->actionFunc = func_80AC9FE4;
+            ActorCutscene_SetIntentToPlay(this2->actor.cutscene);
+            gSaveContext.unk_3DD0[3] = 0;
+        }
+        globalCtx->msgCtx.ocarinaMode = 4;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Time_Tag/func_80ACA3C0.s")
+void func_80ACA12C(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCurrentIndex() != this->actor.cutscene) {
+        this->actionFunc = func_80ACA268;
+        this->actor.textId = 0xC02;
+        Item_Give(globalCtx, ITEM_SONG_SOARING);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Time_Tag/func_80ACA418.s")
+void func_80ACA184(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (ActorCutscene_GetCurrentIndex() == 0x7C) {
+        ActorCutscene_Stop(0x7C);
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    } else if (ActorCutscene_GetCanPlayNext(this->actor.cutscene)) {
+        ActorCutscene_Start(this->actor.cutscene, &this->actor);
+        this->actionFunc = func_80ACA12C;
+    } else {
+        ActorCutscene_SetIntentToPlay(this->actor.cutscene);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Time_Tag/func_80ACA5F8.s")
+void func_80ACA208(EnTimeTag* this, GlobalContext* globalCtx) {
+    if ((Message_GetState(&globalCtx->msgCtx) == 5) && Message_ShouldAdvance(globalCtx)) {
+        func_801477B4(globalCtx);
+        this->actionFunc = func_80ACA268;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Time_Tag/func_80ACA714.s")
+void func_80ACA268(EnTimeTag* this, GlobalContext* globalCtx) {
+    if (Actor_ProcessTalkRequest(&this->actor, &globalCtx->state)) {
+        if (this->actor.textId == 0) {
+            this->actionFunc = func_80ACA184;
+        } else {
+            this->actionFunc = func_80ACA208;
+        }
+    } else if ((this->actor.xzDistToPlayer < 100.0f) && Player_IsFacingActor(&this->actor, 0x3000, globalCtx) &&
+               (Flags_GetSwitch(globalCtx, ENTIMETAG_GET_SWITCHFLAG(&this->actor)) ||
```

Comment:
```suggestion
               Flags_GetSwitch(globalCtx, ENTIMETAG_GET_SWITCHFLAG(&this->actor) ||
```

Reply:
I think this is wrong, because it's combined with another || condition.

Reply:
Oh right, I miscounted the brackets, nvm.